File: src/Varya/Lab7.java
package Varya;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Lab7 {
    public static void main(String[] args) {
        Scanner scanner;
        Scanner sc = new Scanner(System.in);
        System.out.println("input begin symbols");
        String begin = sc.nextLine();
        System.out.println("input end symbols");
        String end = sc.nextLine();
        try {
            scanner = new Scanner(new File("text.txt"));
        } catch (FileNotFoundException e) {
            System.err.println("File not found");
            return;
        }
        while (scanner.hasNextLine()) {
            String[] words = scanner.nextLine().split(" ");
            int max = 0;
            int index = -1;
            for (int i = 0; i < words.length; i++) {
                if (words[i].startsWith(begin) && words[i].endsWith(end) && words[i].length() > max) {
                    max = words[i].length();
                    index = i;
                }
            }
            if (index != -1) {
                StringBuilder line = new StringBuilder();
                for (int i = 0; i < words.length; i++) {
                    if (i != index)
                        line.append(words[i]);
                    line.append(" ");
                }
                System.out.println(line);
            }
        }
    }
}

File: src/lab7/lab7.kt
package lab7

import java.io.File
import java.util.*
import kotlin.collections.ArrayList
import kotlin.collections.HashMap

val checkMap = HashMap<Char, String>()

fun main() {
    val sc = Scanner(File("src/lab7/text"))
    val words = ArrayList<String>()
    while (sc.hasNextLine()) {
        val line = sc.nextLine()
        words.addAll(line.split(" "))
    }

    for (word in words) {
        if (checkMap[word[0]] != null) {
            System.err.println("wrong args")
            return
        } else {
            checkMap[word[0]] = word
        }
    }
    var max = 0
    var first = ""
    for (word in words) {
        val temp = weird(word)
        if (temp > max){
            max = temp
            first = word
        }
    }
    print(first)
    while (checkMap[first.last()] != null){
        if(first == checkMap[first.last()]!!)
            break
        first = checkMap[first.last()]!!
        print(" -> $first")
    }
    println("\n$max")

}
fun weird(word: String): Int {
    if (checkMap[word.last()] == word)
        return 1
    if (checkMap[word.last()] != null)
        return weird(checkMap[word.last()]!!) + 1
    return 1
}
